model/relabel/relabel.go

```diff
@@ -180,8 +176,11 @@ func (re *Regexp) UnmarshalYAML(unmarshal func(interface{}) error) error {
 
 // MarshalYAML implements the yaml.Marshaler interface.
 func (re Regexp) MarshalYAML() (interface{}, error) {
-	if re.original != "" {
-		return re.original, nil
+	if re.String() != "" {
+		str := re.String()
+		str = strings.TrimPrefix(str, "^(?:")
+		str = strings.TrimSuffix(str, ")$")
+		return str, nil
```

Comment:
Since we know the prefix and suffix precisely, couldn't we just write
```Go
    str := re.String()
    return str[4:len(str)-2], nil
```
?

Reply:
`str[4:len(str)-2]` would also probably be slightly more efficient when Marshaling since you wouldn't need to allocate two new strings each time.

Reply:
(1/2) It would be clean to do this in a String() method

Reply:
(Nevermind, strings.TrimPrefix/strings.TrimSuffix doesn't allocate)

Reply:
You're right! And since the fields are unexported we don't have to worry about potential panics.

I've extracted this logic in a String() method, let me know how it looks ^^